test_conformance/images/clReadWriteImage/test_read_1D_buffer.cpp

```diff
@@ -0,0 +1,256 @@
+//
+// Copyright (c) 2023 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "../testBase.h"
+#include <CL/cl.h>
+
+int test_read_image_1D_buffer(cl_context context, cl_command_queue queue,
+                              image_descriptor *imageInfo, MTdata d,
+                              cl_mem_flags flags)
+{
+    int error;
+
+    clMemWrapper image;
+    clMemWrapper buffer;
+
+    // Generate some data to test against
+    BufferOwningPtr<char> imageValues;
+    generate_random_image_data(imageInfo, imageValues, d);
+
+    if (gDebugTrace)
+    {
+        log_info(" - Creating 1D image %d...\n", (int)imageInfo->width);
+        log_info(" with %llu mip levels\n",
+                 (unsigned long long)imageInfo->num_mip_levels);
+    }
+
+    buffer = clCreateBuffer(context, flags, imageInfo->rowPitch, NULL, &error);
+    if (error != CL_SUCCESS)
+    {
+        log_error("ERROR: Unable to create buffer for 1D image buffer of size "
+                  "%d (%s)",
+                  (int)imageInfo->rowPitch, IGetErrorString(error));
+    }
+
+    image = create_image_1d(context, flags, imageInfo->format, imageInfo->width,
+                            0, NULL, buffer, &error);
+    if (image == NULL)
+    {
+        log_error("ERROR: Unable to create 1D image buffer of size %d (%s)",
+                  (int)imageInfo->width, IGetErrorString(error));
+        return -1;
+    }
+
+    if (gDebugTrace) log_info(" - Writing image...\n");
+
+    size_t origin[3] = { 0, 0, 0 };
+    size_t region[3] = { 0, 1, 1 };
```

Comment:
Values in region cannot be 0 as per spec:

"region defines the (width, height, depth) in pixels of the 1D, 2D or 3D rectangle, the (width, height) in pixels of the 2D rectangle and the number of images of a 2D image array or the (width) in pixels of the 1D rectangle and the number of images of a 1D image array. If image is a 2D image object, region[2] must be 1. If image is a 1D image or 1D image buffer object, region[1] and region[2] must be 1. If image is a 1D image array object, region[2] must be 1. The values in region cannot be 0."

Reply:
Good catch, thanks.
This is coming from the refactoring when removing mipmap code from this file.